### ISSUE TITLE
Investigate other package management (e.g. poetry)

### ISSUE THREAD

**Issue by danbraunai** [ApolloResearch/sample#1]
Optionally, also switch to poetry, though I'm unconvinced that the cost of user-friendliness is worth it in general, I'd need to play around with it more.

**Comment by mbalesni** [comment]
I used poetry on my last project and liked the fact that I can add dependencies only once — in my terminal when I install them — and they will automatically be added to the project dependencies. Is that possible to do without poetry? What's the recommended way to do that with just `pip` and `requirements.txt`? I think `pip freeze` dumps the full dependency tree which isn't always desirable.

**Comment by danbraunai** [comment]
Yeah that's pretty nice. The downside might be if you're doing some random dev that requires a package that you don't want to commit, then I guess you have to delete the stuff from your pyproject.toml.

The alternative with pip and requirements is either to just to manually add it, which isn't great if you forget, or use https://pypi.org/project/pipreqs/ which can create requirements based on imports in the project. Can put in pre-commit/CI.

But yeah poetry probably best option. I'll update title of this issue.

**Comment by mbalesni** [comment]
Poetry is compatible with PIP in the sense that you can keep installing packages using PIP for dev and they will work. 

So when I don’t want to commit the package, I install it with:

`pip install X`

And when I want to both install and add it to pyproject.toml I do:

`poetry add X`
